Fix help message for GCS

krishnasrinivas opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
`gcloud init` or `gcloud beta auth application-default login` way of providing credentials in not suitable for applications like minio.

When users want to deploy minio gateway on VMs or using docker containers we can't expect them to do "gloud init" the deployment would have to be automated/scriptable.

https://cloud.google.com/storage/docs/authentication the section Other Environments points to https://developers.google.com/identity/protocols/application-default-credentials#howtheywork which has instructions on how to use GOOGLE_APPLICATION_CREDENTIALS This is most suitable for applications like minio.

I have edited the help message accordingly. We can use the same way for our testing as well, i.e the admin will have to generate and give us credentials.json for testing.

and:

```
export GOOGLE_APPLICATION_CREDENTIALS=/home/krishna/Downloads/miniotest-d4530e7be663.json
minio gateway gcs PROJECTID
```


## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
https://github.com/minio/minio/issues/4391

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

nl5887 committed:
Implement gateway support Google Cloud Storage

nl5887 committed:
Move anonymous error to object translation from Azure specific to gateway

nl5887 committed:
Add comment, gateway should validate object name

nl5887 committed:
Added AllAccessDisabled error

nl5887 committed:
Initial implementation of Google Cloud Storage

nl5887 committed:
Add usage for GCS gateway, closes #4280

nl5887 committed:
Use default endpoint when not supplied

nl5887 committed:
Translate gcs errors to S3 compatible errors, fixes #4278

nl5887 committed:
Allow bucket creation in different regions, closes #4287 and #4241

* I needed to remove the region check from PutBucketHandler

nl5887 committed:
Return BucketAlreadyExists when bucket exists with another user

nl5887 committed:
Make every backend responsible for parsing its own arguments, fixes #4293

nl5887 committed:
Map S3 regions to Google (multi)regions

nl5887 committed:
Return correct error when PutObject fails

nl5887 committed:
Implement bucket policies

nl5887 committed:
Add access and secret key to example, needed to access Minio Gateway

nl5887 committed:
Use MINIO_REGION environment variable for region configuration, closes #4287

nl5887 committed:
Support marker, closes #4286

nl5887 committed:
Use maxKeys for iterator

nl5887 committed:
Verify md5 content hash, closes #4285

nl5887 committed:
Add region to gcs gateway example

nl5887 committed:
Support iterating through ListObjectParts using NextPartNumberMarker, closes #4284

nl5887 committed:
Verify multipart etag during complete, closes #4288

nl5887 committed:
Fix issue with UNSIGNED payloads.

Additionally also fixes escaping slashes in
temporary multipart names

nl5887 committed:
Fix issue with AbortMultipartUpload, closes #4322

nl5887 committed:
Fix uploadIDMarker handling.

nl5887 committed:
Fix ListObjectParts to list properly all parts - closes #4322

nitisht committed:
Fixes https://github.com/minio/minio/issues/4320 (#4332)

- Add description for error ErrBucketAlreadyExists

vadmeste committed:
gateway: Use default params when no args provided (#4315)

For S3 & Azure, use default parameters when no arguments (endpoint) are
provided. This also avoids a crash.

harshavardhana committed:
gateway/gcs: send proper error responses for Get/SetBucket policies. (#4338)

Fixes #4323

nl5887 committed:
Map only default region us-east-1 to gcs us region

nl5887 committed:
Cleanup and update the PR with the master branch.

Commit:
Change md5Sum to etag (#4399)

poornas committed:
Remove profile option for gcs from gateway help message (#4421)

krishnasrinivas committed:
gateway-gcs: Help message to use GOOGLE_APPLICATION_CREDENTIALS

fixes #4391

vadmeste committed:
fix: Check project id before starting gateway (#4412)

balamurugana reviewed (commented):
What will happen if `gcloud init --console-only` is used instead of `GOOGLE_APPLICATION_CREDENTIALS`?

krisis commented on cmd/gateway-main.go:
```diff
@@ -63,7 +63,8 @@ EXAMPLES:
 
   4. Start minio gateway server for GCS backend.
-      $ gcloud init --console-only # this will configure your google account
+      $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
```
What a user needs to do for Minio GCS gateway to successfully authenticate with the GCS account depends on the environment it is being run. This message applies only when the gateway is run _outside_ GCP. Is it possible for the gateway to know if it's running in a GCP instance? If not, we could modify the message to,

> (When run outside Google Cloud Platform)
$ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json

krishnasrinivas replied:
Not true @krisis this step works everywhere

poornas committed:
Set default ETag value if vendor returns empty md5 string (#4409)

The ETag is constructed from md5 atttribute of object attributes
returned by the vendor's Composer. The md5 attribute comes back
as nil for large uploads. Instead the CRC32C should be used.

Refer to https://cloud.google.com/storage/docs/hashes-etags

Fixes #4397
